tidy3d/components/heat/source.py

```diff
@@ -0,0 +1,48 @@
+"""Defines heat material specifications"""
+from __future__ import annotations
+
+from abc import ABC
+from typing import Union, Tuple
+
+import pydantic as pd
+
+from .viz import plot_params_heat_source
+
+from ..base import Tidy3dBaseModel, cached_property
+from ..data.data_array import TimeDataArray
+from ..viz import PlotParams
+
+from ...constants import VOLUMETRIC_HEAT_RATE
+
+
+class HeatSource(ABC, Tidy3dBaseModel):
+    """Abstract heat source."""
+
+    structures: Tuple[str] = pd.Field(
```

Comment:
I might be mistaken but I think `Tuple[str]` means that it has to be a tuple of only **one** string, eg `("hello",)`.  I think what you might want is `Tuple[str, ...]` but double check me on this.